internal/models/task_id_test.go

```diff
@@ -0,0 +1,145 @@
+package models_test
+
+import (
+	"testing"
+
+	"github.com/formancehq/payments/internal/models"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTaskID(t *testing.T) {
+	t.Parallel()
+
+	t.Run("String", func(t *testing.T) {
+		t.Parallel()
+		// Given
+		
+		id := models.TaskID{
+			Reference: "task123",
+			ConnectorID: models.ConnectorID{
+				Provider:  "stripe",
+				Reference: uuid.MustParse("00000000-0000-0000-0000-000000000001"),
+			},
+		}
+		
+		// When
+		result := id.String()
+		
+		// Then
+		assert.NotEmpty(t, result)
+	})
+
+	t.Run("TaskIDFromString", func(t *testing.T) {
+		t.Parallel()
+		// Given
+		
+		original := models.TaskID{
+			Reference: "task123",
+			ConnectorID: models.ConnectorID{
+				Provider:  "stripe",
+				Reference: uuid.MustParse("00000000-0000-0000-0000-000000000001"),
+			},
+		}
+		
+		idStr := original.String()
+		
+		// When
+		id, err := models.TaskIDFromString(idStr)
+		
+		// Then
+		require.NoError(t, err)
+		assert.Equal(t, original.Reference, id.Reference)
+		assert.Equal(t, original.ConnectorID.Provider, id.ConnectorID.Provider)
+		assert.Equal(t, original.ConnectorID.Reference.String(), id.ConnectorID.Reference.String())
+		
+		_, err = models.TaskIDFromString("invalid-base64")
+		// Then
+			assert.Error(t, err)
```

Comment:
test the error message too. Take inspiration from account_id_test.go file in this PR